src/physlr-molecules.cc

```diff
@@ -450,18 +529,66 @@ main(int argc, char* argv[])
 	}
 
 	auto vertexItRange = vertices(g);
-	for (auto vertexIt = vertexItRange.first; vertexIt != vertexItRange.second; ++vertexIt) {
-		componentToVertexSet_t componentsVec;
-		vertexToComponent_t vertexToComponent;
 
-		// Find neighbour of vertex and generate neighbour induced subgraph
-		auto neighbours = boost::adjacent_vertices(*vertexIt, g);
+	bool openmp = false;
+#if _OPENMP
+	openmp = true;
+#endif
+
+	if (threads > 1 && openmp) {
```

Comment:
```suggestion
	if (threads > 1) {
```
Why do you need all that extra stuff?

Reply:
You mean the `if-else` statement?
so when using multiple threads we need to make an auxiliary data structure that needs time and memory. When open mp is disabled or we are using only one thread we can skip using that data-structure and do the job faster (compared to 1-thread and using the aux data structure) and with ~ 30% less ram usage.

Reply:
Right, but why do you need the `openmp` variable.

Reply:
Mmm, what if they compile without `openmp` but they set `-t10`?
it'd be better to output a warning message instead of this. hum?

Reply:
you can do 
```

#if ! _OPENMP
if (threads > 1) {
threads = 1;
std::cerr << ""Setting threads to 1" <<std::endl
}
#endif
 
--

Reply:
I dont think they can even compile without openMP btw. We have that compilation flag on in the make file. It's also a requirement on the README.